bench/txdb.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const random = require('bcrypto/lib/random');
+const bench = require('./bench');
+const WalletDB = require('../lib/wallet/walletdb');
+const MTX = require('../lib/primitives/mtx');
+const Input = require('../lib/primitives/input');
+const Output = require('../lib/primitives/output');
+const Script = require('../lib/script/script');
+const {rimraf, testdir} = require('../test/util/common');
+
+const ITERATIONS = 1000;
+
+(async () => {
+  for (const memory of [true, false]) {
+    console.log(`Memory: ${memory}`);
+
+    const location = testdir('txdb');
+
+    // Create walletDB and wallet
+    const wdb = new WalletDB({
+      memory,
+      location
+    });
+    await wdb.open();
+    const wallet = await wdb.create();
+    const addr = await wallet.receiveAddress();
+    const script = Script.fromAddress(addr);
+
+    {
+      // Fund wallet with 1,000 BTC in single UTXO
+      const fund = new MTX();
+      const input = new Input();
+      // Make this a non-coinbase TX so we can spend right away
+      input.prevout.hash = random.randomBytes(32);
+      fund.inputs.push(input);
+      const output = new Output();
+      output.value = 1000e8;
+      output.script = script;
+      fund.outputs.push(output);
+
+      // Confirm
+      const dummyBlock = {
+        height: 0,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wdb.addTX(fund.toTX(), dummyBlock);
+    }
+
+    // Self-send small TXs so txdb is filled
+    // with both spent and unspent coins
+    console.log(' Filling txdb...');
+    for (let i = 1; i <= ITERATIONS; i++) {
+      const tx = await wallet.send({
+        outputs: [{
+          address: addr,
+          value: 1000
+        }]
+      });
+      const dummyBlock = {
+        height: i,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wdb.addTX(tx, dummyBlock);
+    }
+
+    // Now send entire balance, forcing the wallet
+    // to spend all its coins. In this test we can ignore
+    // size limits, so enormous amount of inputs is OK.
+    const {unconfirmed} = await wallet.getBalance();
+    const mtx = new MTX();
+    const output = new Output();
+    output.value = unconfirmed;
+    output.script = script;
+    mtx.outputs.push(output);
```

Comment:
Do we still need this?

Reply:
...or does this need to be updated now that coin selection was merged?

Reply:
This script was written to see if the `u` bucket addition is actually an improvement.